src/vtarget-out/inv-syn/inv_syn_cegar.cc

```diff
@@ -0,0 +1,962 @@
+    if (! fin.is_open()) {
+        ILA_ERROR << "Unable to read the synthesis result file:" << synthesis_result_fn;
+        status = cegar_status::FAILED;
+        bad_state = true;
+        return true; // reachable
+    } 
+    sbuf << fin.rdbuf();
+    cex_reachable = !(S_IN("proved",sbuf.str()));
+    if (S_IN("unknown",sbuf.str())) cex_reachable = true;
+    // count cands
+    total_grain_cand += retrieveColonEol (sbuf.str(), "TotalCand:");
+  } else {
+    std::string line;
+    { // read the result
+      std::ifstream fin(synthesis_result_fn);
+      if (! fin.is_open()) {
+        ILA_ERROR << "Unable to read the synthesis result file:" << synthesis_result_fn;
+        status = cegar_status::FAILED;
+        bad_state = true;
+        return true; // reachable
+      } 
+      std::getline(fin,line);  
+    } // finish file reading
+    cex_reachable = true;
+    if (S_IN("unsat", line))
+      cex_reachable = false; // not reachable
+  }
+  // else reachable
+  return cex_reachable;
+}
+
+
+// -------------------------------- MISCS ------------------------------------------- //
+
+void InvariantSynthesizerCegar::LoadDesignSmtInfo(const std::string & fn) {
+  std::ifstream fin(fn);
+  if (! fin.is_open()) {
+    ILA_ERROR << "Unable to read from : " << fn;
+    return;
+  }
+  std::stringstream sbuf;
+  sbuf << fin.rdbuf();
+  design_smt_info = std::make_shared<smt::YosysSmtParser>(sbuf.str());
+}
+
+const std::vector<std::string> & InvariantSynthesizerCegar::GetRunnableTargetScriptName() const {
+  return runnable_script_name;
+}
+
+void InvariantSynthesizerCegar::LoadPrevStatisticsState(const std::string & fn) {
+  DesignStatistics local_info;
+  local_info.LoadFromFile(fn);
+  if ( eqcheck_time != 0 || 
+      inv_proof_attempt_time != 0 ||
+      inv_syn_time != 0 ||
+      inv_validate_time != 0 ||
+      inv_enhance_time != 0 ||
+      ! inv_syn_time_series.empty() ) {
+    ILA_ERROR << "Not loading the statistics state from initial time!";
+    }
+  eqcheck_time += local_info.TimeOfEqCheck;
+  inv_proof_attempt_time += local_info.TimeOfInvProof;
+  inv_syn_time += local_info.TimeOfInvSyn;
+  inv_enhance_time += local_info.TimeOfInvSynEnhance;
+  inv_validate_time += local_info.TimeOfInvValidate;
+  inv_syn_time_series.insert(inv_syn_time_series.end(),local_info.TimeOfInvSynSeries.begin(), local_info.TimeOfInvSynSeries.end());
+}
+
+DesignStatistics InvariantSynthesizerCegar::GetDesignStatistics() const {
+  DesignStatistics ret;
+
+  ret.TimeOfEqCheck       = eqcheck_time;
+  ret.TimeOfInvProof      = inv_proof_attempt_time;
+  ret.TimeOfInvSyn        = inv_syn_time;
+  ret.TimeOfInvSynEnhance = inv_enhance_time;
+  ret.TimeOfInvValidate   = inv_validate_time;
+  ret.TimeOfInvSynSeries  = inv_syn_time_series;
+
+  if (design_smt_info == nullptr) {
+    ILA_ERROR << "Design information not available!";
+    return ret;
+  }
+  std::string query_mod_name = implementation_top_module_name;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'query_mod_name' is assigned a value that is never used.](https://app.codacy.com/app/Bo-Yuan-Huang/ILAng/pullRequest?prid=4555809)